ValueError: array must not contain infs or NaNs

Add alert on according level of analytics with user friendly message

```
2019-04-12 19:48:24,346 [Analytics] [ERROR]  handle_analytic_task Exception: 'Traceback (most recent call last):
  File "bin/../analytics/analytic_unit_manager.py", line 82, in handle_analytic_task
    result_payload = await self.__handle_analytic_task(task)
  File "bin/../analytics/analytic_unit_manager.py", line 69, in __handle_analytic_task
    return await worker.do_train(payload['segments'], data, payload['cache'])
  File "bin/../analytics/analytic_unit_worker.py", line 41, in do_train
    new_cache: ModelCache = await asyncio.wait_for(self._training_future, timeout = config.LEARNING_TIMEOUT)
  File "/usr/lib64/python3.6/asyncio/tasks.py", line 358, in wait_for
    return fut.result()
  File "/usr/lib64/python3.6/concurrent/futures/thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
  File "bin/../analytics/detectors/pattern_detector.py", line 49, in train
    new_cache = self.model.fit(dataframe, segments, self.analytic_unit_id, cache)
  File "bin/../analytics/models/model.py", line 77, in fit
    segment = Segment(dataframe, segment_map, self.find_segment_center)
  File "bin/../analytics/models/model.py", line 23, in __init__
    self.center_index = center_finder(dataframe, self.start, self.end)
  File "bin/../analytics/models/drop_model.py", line 38, in find_segment_center
    segment_center_index = utils.find_pattern_center(segment, start, 'drop')
  File "bin/../analytics/utils/common.py", line 346, in find_pattern_center
    segment_median = utils.get_distribution_density(segment_data)[0]
  File "bin/../analytics/utils/common.py", line 316, in get_distribution_density
    pdf = gaussian_kde(segment)
  File "/usr/lib64/python3.6/site-packages/scipy/stats/kde.py", line 172, in __init__
    self.set_bandwidth(bw_method=bw_method)
  File "/usr/lib64/python3.6/site-packages/scipy/stats/kde.py", line 499, in set_bandwidth
    self._compute_covariance()
  File "/usr/lib64/python3.6/site-packages/scipy/stats/kde.py", line 510, in _compute_covariance
    self._data_inv_cov = linalg.inv(self._data_covariance)
  File "/usr/lib64/python3.6/site-packages/scipy/linalg/basic.py", line 946, in inv
    a1 = _asarray_validated(a, check_finite=check_finite)
  File "/usr/lib64/python3.6/site-packages/scipy/_lib/_util.py", line 238, in _asarray_validated
    a = toarray(a)
  File "/usr/lib64/python3.6/site-packages/numpy/lib/function_base.py", line 1233, in asarray_chkfinite
    "array must not contain infs or NaNs")
ValueError: array must not contain infs or NaNs
'
```

--- a/analytics/analytics/utils/common.py
+++ b/analytics/analytics/utils/common.py
@@ -309,6 +309,7 @@ def get_correlation(segments: list, av_model: list, data: pd.Series, window_size
     return correlation_list
 
 def get_distribution_density(segment: pd.Series) -> float:
+    segment.dropna(inplace = True)
     if len(segment) < 2 or len(segment.nonzero()[0]) == 0:
         return (0, 0, 0)
     min_jump = min(segment)

--- a/analytics/tests/test_utils.py
+++ b/analytics/tests/test_utils.py
@@ -195,6 +195,12 @@ class TestUtils(unittest.TestCase):
         utils_result_segment = utils.get_distribution_density(segment)
         self.assertEqual(len(utils_result_data), 3)
         self.assertEqual(utils_result_segment, (0, 0, 0))
+
+    def test_get_distribution_density_with_nans(self):
+        segment = [np.NaN, 1, 1, 1, np.NaN, 3, 5, 5, 5, np.NaN]
+        segment = pd.Series(segment)
+        result = (3, 5, 1)
+        self.assertEqual(utils.get_distribution_density(segment), result)
     
     def test_find_pattern_jump_center(self):
         data = [1.0, 1.0, 1.0, 5.0, 5.0, 5.0]

